### ISSUE TITLE
suspend/resume breaks when you change a route

### ISSUE THREAD

**Issue by flyingmachine** [duct-framework/server.http.jetty#3]
I'm working with the example app and found that if you suspend the system, change the config by removing the example route, and resume the system, the HTTP server will stop and not restart. To repro:

```clojure
;; in repl
(require
  '[integrant.core :as ig]
  '[integrant.repl :as ir]
  '[duct.core :as duct]
  '[clojure.java.io :as io])

(ir/set-prep! #(duct/prep (duct/read-config (io/resource "duct_9/config.edn"))))
(ir/go)
(ir/suspend)
;; modify config.edn, removing example route
(ir/resume)
```

Here's what my config.edn looks like after I edit it:

```clojure
{:duct.core/project-ns  duct-9
 :duct.core/environment :production

 :duct.module/logging {}
 :duct.module.web/site {}
 
 :duct.router/cascading
 []}
```

**Comment by flyingmachine** [comment]
It looks like [integrant.core/resume halts the server](https://github.com/weavejester/integrant/blob/master/src/integrant/core.cljc#L298), but [resume-key thinks the config hasn't changed](https://github.com/duct-framework/server.http.jetty/blob/master/src/duct/server/http/jetty.clj#L30) so it doesn't restart it

**Comment by weavejester** [comment]
It looks like this is an issue with Integrant. What we probably want to do is change this line:

```clojure
(halt! system (missing-keys system keys))
```

Which halts all missing keys *and* the keys that depend on them. Instead, it should probably be something more like:

```clojure
(halt! (apply dissoc system (missing-keys system keys)))
```

But I'll need to think about the consequences and make sure I have the logic right.

**Comment by weavejester** [comment]
This should be fixed by Integrant 0.4.1. I've updated the dependencies for server.http.jetty to include that, and released server.http.jetty 0.1.5.

**Comment by flyingmachine** [comment]
awesome! so far so good :)
